src/core/l10n.js

```diff
@@ -171,6 +171,22 @@ l10n["zh-cn"] = l10n.zh;
 
 export const lang = html && html.lang in l10n ? html.lang : "en";
 
+/**
+ * @typedef {Record<string, Record<string, string|Function>>} LanguageData
+ * @type {<T extends LanguageData>(obj: T) => T[keyof T]}
```

Comment:
Can we use `@template`, `@param` and `@return` instead?

Reply:
@saschanaz Can you help me with this one? Specifically the `T extends` part?

Reply:
AFAIK:

```suggestion
 * @template {Languagedata} T
 * @param {T} obj
 * @return {T[keyof T]}
```

Reply:
This is same as what I came up with, but:
![image](https://user-images.githubusercontent.com/8426945/71074774-df222280-21a8-11ea-8162-50e5a086106d.png)

Reply:
Looks like narrowing the type isn't supported: https://github.com/Microsoft/TypeScript/issues/1178#issuecomment-348755243

Reply:
```suggestion
 */
/**
 * @template {LanguageData} T
 * @param {T} localizationStrings
 * @return {T[keyof T]}
```

No idea why but splitting works 🤣

Reply:
:rofl: :shipit: